specs/core/0_beacon-chain.md

```diff
@@ -1340,20 +1383,17 @@ def process_block_header(state: BeaconState, block: BeaconBlock) -> None:
     # Verify that the slots match
     assert block.slot == state.slot
     # Verify that the parent matches
-    assert block.parent_root == signing_root(state.latest_block_header)
+    assert block.parent_root == hash_tree_root(state.latest_block_header)
     # Save current block as the new latest block
     state.latest_block_header = BeaconBlockHeader(
         slot=block.slot,
         parent_root=block.parent_root,
         # `state_root` is zeroed and overwritten in the next `process_slot` call
         body_root=hash_tree_root(block.body),
-        # `signature` is zeroed
     )
     # Verify proposer is not slashed
     proposer = state.validators[get_beacon_proposer_index(state)]
     assert not proposer.slashed
-    # Verify proposer signature
-    assert bls_verify(proposer.pubkey, signing_root(block), block.signature, get_domain(state, DOMAIN_BEACON_PROPOSER))
```

Comment:
Just wondering, why is this assertion removed?

Reply:
The signature check is moved to a separate function, `verify_block_signature`, to separate the actual header processing from the signature check that clients would perform in their own optimal way. Now that we have the clear distinction between signed and unsigned header data, I think this is cleaner. And functionally it's still the same.

Reply:
Sounds good. Just wanted to ask as a sanity check here since the diff is hard to follow. Thanks for the response.